pkg/resources/catalog.cattle.io/v1/clusterrepo/validator_test.go

```diff
@@ -0,0 +1,131 @@
+package clusterrepo
+
+import (
+	"context"
+	"encoding/json"
+	"testing"
+
+	catalogv1 "github.com/rancher/rancher/pkg/apis/catalog.cattle.io/v1"
+	"github.com/rancher/webhook/pkg/admission"
+	"github.com/stretchr/testify/assert"
+	admissionv1 "k8s.io/api/admission/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+func TestClusterRepoValidation(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
```

Comment:
I think you're still missing some test cases. 
You have four conditions (both missing, only GitRepo is provided, only ClusterRepo is provided, both are provided) and two operations (create and update). This makes 8 cases.

Reply:
Done. Added 8 cases